qiskit/visualization/circuit/latex.py

```diff
@@ -85,52 +103,15 @@ def __init__(
         Raises:
             ImportError: If pylatexenc is not installed
         """
-        if qregs is not None:
-            warn(
-                "The 'qregs' kwarg to the QCircuitImage class is deprecated "
-                "as of 0.20.0 and will be removed no earlier than 3 months "
-                "after the release date.",
-                DeprecationWarning,
-                2,
-            )
-        if cregs is not None:
-            warn(
-                "The 'cregs' kwarg to the QCircuitImage class is deprecated "
-                "as of 0.20.0 and will be removed no earlier than 3 months "
-                "after the release date.",
-                DeprecationWarning,
-                2,
-            )
-        if layout is not None:
-            warn(
-                "The 'layout' kwarg to the QCircuitImage class is deprecated "
-                "as of 0.20.0 and will be removed no earlier than 3 months "
-                "after the release date.",
-                DeprecationWarning,
-                2,
-            )
-        if global_phase is not None:
-            warn(
-                "The 'global_phase' kwarg to the QCircuitImage class is deprecated "
-                "as of 0.20.0 and will be removed no earlier than 3 months "
-                "after the release date.",
-                DeprecationWarning,
-                2,
-            )
+        del layout
+        del global_phase
         # This check should be removed when the 4 deprecations above are removed
         if circuit is None:
-            warn(
-                "The 'circuit' kwarg to the QCircuitImage class must be a valid "
-                "QuantumCircuit and not None. A new circuit is being created using "
-                "the qubits and clbits for rendering the drawing.",
-                DeprecationWarning,
-                2,
-            )
             circ = QuantumCircuit(qubits, clbits)
-            for reg in qregs:
+            for reg in qregs or []:
```

Comment:
This was a bug fix. If `qregs` is None, then you can't iterate over it.

Reply:
that's true, but passing neither `qregs` nor `circuit` is an invalid calling convention, so it's fine for it just to explode tbh.  (It doesn't come up because this is actually just an internal-detail class.)